heat/core/arithmetics.py

```diff
@@ -575,6 +1027,33 @@ def diff(
     return ret
 
 
+def diff_(a: DNDarray, n: int = 1, axis: int = -1) -> DNDarray:
+    """
+    In-place version of `diff`. Not implemented as this is not an element-wise operation and there
+    is no in-place version of `diff` in PyTorch.
+
+    Calculate the n-th discrete difference along the given axis.
+    The first difference is given by ``out[i]=a[i+1]-a[i]`` along the given axis, higher differences
+    are calculated by using diff_ recursively. The shape of the output is the same as ``a`` except
+    along axis where the dimension is smaller by ``n``. The datatype of the output is the same as
+    the datatype of the difference between any two elements of ``a``. The split does not change. The
+    output array is balanced.
+
+    Parameters
+    -------
+    a : DNDarray
+        Input array
+    n : int, optional
+        The number of times values are differenced. If zero, the input is returned as-is.
+        ``n=2`` is equivalent to ``diff_(diff_(a))``
+    axis : int, optional
+        The axis along which the difference is taken, default is the last axis.
+    """
+    raise NotImplementedError(
+        "Not implemented as this is not an element-wise operation and there is no in-place version of `diff` in PyTorch."
+    )
```

Comment:
As I added in-place versions to almost all arithmetic operations, I thought it might make sense to add NotImplementsErrors for the remaining operations which don't have an in-place version. Therefore, I scanned the existing code to find examples for not implemented functions which throw such errors (e.g. `fit` in `heat/core/base.py`) and did it the same way. Also I tried to explain why specified functions like `diff_` here don't work in order to spare future contributors some time wondering why we have in-place version for all arithmetic operations except for a hand full of them.
If you say this is unnecessary and it should be self-explanatory enough, why there is an `add_` but not a `diff_` function, I can delete all of that of course.